c7n/resources/ami.py

```diff
@@ -294,17 +302,32 @@ class ImageUnusedFilter(Filter):
                 filters:
                   - type: unused
                     value: true
+                    include-launch-configurations: true
     """
 
-    schema = type_schema('unused', value={'type': 'boolean'})
+    schema = type_schema(
+        'unused',
+        value={'type': 'boolean'},
+        **{
+            'include-launch-configurations': {'type': 'boolean'}
+        }
```

Comment:
not clear why this is being parameterized, the intent was already present re default to pulling templates.

Reply:
I think the parameter here could be named better, the original ask was to include all launch configs, whether or not there's an associated autoscaling group to the launch configuration.

> When I create a Custodian policy to identify old and unused AMIs, it does not check if the AMI is referenced in a Launch Config or Launch Template. We deregistered an AMI that a team was using for ephemeral workloads due to this issue, which caused some production downtime.

Reply:
ic.. so technically this not related to asg, this is just arbitrary launch templates that might be in use but not referenced. feels like we need to better distinguish the name, if we're also checking asg ref'd launch configs by default.

Reply:
also this implementation has an assumption of asg associated launch templates.

Reply:
Yea, thinking of maybe `unused-configs-templates` or `include-unused-configs-templates`